test/e2e/validators.go

```diff
@@ -50,6 +49,24 @@ func cleanupOptions(ctx *framework.TestCtx) *framework.CleanupOptions {
 	}
 }
 
+// clientGet is a wrapper around f.Client.Get that performs retries in case of retryable errors
+func clientGet(key types.NamespacedName, obj runtime.Object) error {
+	f := framework.Global
+
+	return wait.PollImmediate(1*time.Second, 60*time.Second, func() (bool, error) {
+		err := f.Client.Get(goctx.TODO(), key, obj)
+
+		if err == nil {
+			return true, nil
+		}
+		if apierrors.IsServerTimeout(err) || apierrors.IsTimeout(err) || apierrors.IsTooManyRequests(err) {
```

Comment:
If I understand correctly:
- IsServerTimeout is not needed, because this only indicates that a request needs to be tried, if our `f.Client.Get(goctx.TODO(), key, obj)` returns an error, we will retry the request automatically.
- IsTimeout should be convered by the PollInmediate timeout.
- IsTooManyRequests, this I think is a good one. Can we have this one inside the oneliner I mentioned above, and if this is TRUE, modify the error we returned with more info? e.g. "Server cannot handle too many requests at the moment"

Reply:
@qu1queee no, the idea of my code is to retry only selective errors and those are servertimeout, timeout and toomanyrequests. All other errors (like not found), I am returning immediately. And this imo makes sense because we have test cases that do a get to verify that something is gone.